src/configurationProvider.ts

```diff
@@ -88,29 +88,27 @@ export class JavaDebugConfigurationProvider implements vscode.DebugConfiguration
                 await updateDebugSettings();
             }
 
-            if (Object.keys(config).length === 0) { // No launch.json in current workspace.
-                // check whether it is opened as a folder
-                if (folder !== undefined) {
-                    // for opened with folder, return directly.
-                    return config;
-                }
-                // only rebuild for single file case before the build error issue is resolved.
+            // just create and open the lanuch.json if there is no lanch.json in current folder
+            if (Object.keys(config).length === 0 && folder !== undefined) {
+                return config;
+            }
+
+            // generate config in memory if there is noly single file and no launch.json

```

Comment:
If it's the single file case that no workspace folder is opened, generate debug config in memory.